daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Recursion.scala

```diff
@@ -6,59 +6,26 @@ package com.daml.lf.validation
 import com.daml.lf.data.Ref._
 import com.daml.lf.language.Ast._
 import com.daml.lf.language.Graphs
-import com.daml.lf.validation.traversable.{ExprTraversable, TypeTraversable}
+import com.daml.lf.validation.traversable.TypeTraversable
 
 private[validation] object Recursion {
 
   /* Check there are no cycles in the module references */
 
   @throws[ValidationError]
-  def checkPackage(pkgId: PackageId, modules: Map[ModuleName, Module]): Unit = {
-    val g = modules.map {
-      case (name, mod) => name -> (mod.definitions.values.flatMap(modRefs(pkgId, _)).toSet - name)
+  def checkPackage(world: World, pkgId: PackageId, modules: Map[ModuleName, Module]): Unit = {
+    val g = modules.transform { (name, _) =>
+      def context = ContextModule(pkgId, name)
+      world.idsInModule(context, pkgId, name).collect {
+        case Identifier(`pkgId`, QualifiedName(modName, _)) => modName
```

Comment:
What do the backticks do here? Turn an expression into a pattern that matches exactly the value of the expression?

Reply:
Yes